x/txreward/internal/keeper/keeper.go

```diff
@@ -0,0 +1,234 @@
+package keeper
+
+import (
+	"github.com/commercionetwork/commercionetwork/x/txreward/internal/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/distribution"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/staking/exported"
+)
+
+type Keeper struct {
+	StoreKey sdk.StoreKey
+
+	BankKeeper         bank.Keeper
+	StakeKeeper        staking.Keeper
+	DistributionKeeper distribution.Keeper
+
+	Cdc *codec.Codec
+}
+
+func NewKeeper(storeKey sdk.StoreKey, bk bank.Keeper, sk staking.Keeper, dk distribution.Keeper, cdc *codec.Codec) Keeper {
+	return Keeper{
+		StoreKey:           storeKey,
+		BankKeeper:         bk,
+		StakeKeeper:        sk,
+		DistributionKeeper: dk,
+		Cdc:                cdc,
+	}
+}
+
+func (k Keeper) getFundersStoreKey() []byte {
+	return []byte(types.BlockRewardsPoolFundersPrefix)
+}
+
+func (k Keeper) setFunders(ctx sdk.Context, updatedFunders types.Funders) {
+	store := ctx.KVStore(k.StoreKey)
+	store.Set(k.getFundersStoreKey(), k.Cdc.MustMarshalBinaryBare(&updatedFunders))
+}
+
+//Add new funder to the list of the authorized funders of the block rewards pool
+func (k Keeper) AddBlockRewardsPoolFunder(ctx sdk.Context, funder types.Funder) {
+	var funders types.Funders
+	store := ctx.KVStore(k.StoreKey)
+
+	fundersBz := store.Get(k.getFundersStoreKey())
+	if fundersBz == nil {
+		funders = types.Funders{}.AppendFunderIfMissing(funder)
+	} else {
+		k.Cdc.MustUnmarshalBinaryBare(fundersBz, &funders)
+		funders = funders.AppendFunderIfMissing(funder)
+	}
+	k.setFunders(ctx, funders)
+}
+
+//Return all the block rewards pool's funders
+func (k Keeper) GetBlockRewardsPoolFunders(ctx sdk.Context) types.Funders {
```

Comment:
Do we really want to block people into depositing inside the pool of funds that can be used to give out TBRs? 

I think that we should avoid limiting the deposits so that anyone can deposit any amount at any time he wants. We might want this feature in order to promote community contributions or allow users to do some "charity" to the chain system itself.